Resolve Typescript Issues 1/2

jwatts777 opened the pull request:
## Linked Issue

Closes #2115 

## Description

Cleanup and remove `any` types from documentation sections

## Changsets

Instructions: Changesets automate our changelog. If you modify files in `/packages/skeleton`, run `pnpm changeset` in the root of the monorepo, follow the prompts, then commit the markdown file. Changes that add features should be `minor` while chores and bugfixes should be `patch`. Please prefix the changeset message with `feat:`, `bugfix:` or `chore:`.

## Checklist

Please read and apply all [contribution requirements](https://www.skeleton.dev/docs/contributing).

- [x] This PR targets the `dev` branch (NEVER `master`)
- [x] Documentation reflects all relevant changes
- [x] Branch is prefixed with: `docs/`, `feat/`, `chore/`, `bugfix/`
- [x] Ensure Svelte and Typescript linting is current - run `pnpm check`
- [x] Ensure Prettier linting is current - run `pnpm format`
- [x] All test cases are passing - run `pnpm test`
- [x] Includes a changeset (if relevant; see above)

jwatts777 commented:
[This is the most updated list](https://github.com/skeletonlabs/skeleton/issues/2115#issuecomment-1751511685)
Calling it good for tonight.

Mahmoud-zino commented on sites/skeleton.dev/src/lib/components/DocsIcon/icons.ts:
```diff
@@ -3,8 +3,18 @@
+    path: string;
+};
+
+export type Icons = {
```
@jwatts777 please add `[key: string]: Icon;` here so we can index `Icons` using strings. this solves 2 of svelte-check errors.

Mahmoud-zino commented on sites/skeleton.dev/src/lib/themes.ts:
```diff
@@ -1,7 +1,22 @@
 
-const ghPath = 'https://github.com/skeletonlabs/skeleton/blob/master/src/themes';
-export const themes: any[] = [
+interface Font {
```
We are missing a `url:string` here, the url is used as href for the buttons in the theme.css.
we should also provide a url for each font in `themes`.
this solves another 2 of the svelte-check errors.

jwatts777 replied:
Still waiting on more explanation.

jwatts777 replied:
Tentative change in place.

Mahmoud-zino commented on sites/skeleton.dev/src/routes/(inner)/elements/forms/+page.svelte:
```diff
@@ -7,6 +7,18 @@
+    disabled?: boolean;
+    multiple?: boolean;
+    autocomplete?: string;
+    tabindex?: string;
```
this should be `tabindex?: number;` instead.
the same change must be applied in `/elements/forms/settings.ts`

jwatts777 replied:
If I implement this change, line 16 needs a change as well

jwatts777 reviewed (commented):
Is there something here that I am missing?

endigo9740 commented:
@jwatts777 FYI you can leave these unticked when creating the issue, then tap the generated UI checkboxes afterwards. They are interactive. Much simpler and the proper markdown formatting will be applied automatically.

![image](https://github.com/skeletonlabs/skeleton/assets/1509726/f5ce1250-6383-4701-8885-724655eccca2)

AdrianGonz97 reviewed (changes requested):
Looks good! I've made a couple of changes to fix some types, specifically those that involve `sveld`, rearranged the import order of some components, fixed some typos (these were not introduced in this PR, but we may as well squash them when noticed) and fixed a small logic bug with the attribute mapper. 

I've also added suggestions below to tidy up the `Theme` type so that we're not redefining it's type in 2 separate locations.

Thanks!

jwatts777 commented on sites/skeleton.dev/src/routes/(inner)/docs/get-started/SectionThemes.svelte:
```diff
@@ -14,8 +13,23 @@ plugins: [
 ]\n
 `;
 
+	interface Font {
+		source: string;
+		name: string;
+		file: string;
+		import: string;
+	}
+
+	interface ActiveTheme {
+		file: string;
+		name: string;
+		fonts: Font[];
+		colors: string[];
+		surface: string;
+	}
+
```
Thank you for doing this. I had the same idea, but was nervous about being proactive in following DRY principles.

jwatts777 commented:
> Looks good! I've made a couple of changes to fix some types, specifically those that involve `sveld`, rearranged the import order of some components, fixed some typos (these were not introduced in this PR, but we may as well squash them when noticed) and fixed a small logic bug with the attribute mapper.
> 
> I've also added suggestions below to tidy up the `Theme` type so that we're not redefining it's type in 2 separate locations.
> 
> Thanks!

Just for my own info, you returned a secondary wrapping of the return array with Object.entries vs Object.keys? 
Why do we do this instead of just returning back the object? We're grabbing, wrapping in an array, then returning a map of that array.

AdrianGonz97 commented:
> Just for my own info, you returned a secondary wrapping of the return array with Object.entries vs Object.keys?
Why do we do this instead of just returning back the object? We're grabbing, wrapping in an array, then returning a map of that array.

@jwatts777 

`Object.keys` was only returning the `key` string and not the `value`, so if you look at the deployment before the fix [here](https://skeleton-docs-ayur8fcnj-skeleton-labs.vercel.app/elements/forms#input), and view the code for `Input`, you'll see that the first two characters of each key were being applied:
![img](https://i.imgur.com/l1a6t3d.png)

when it should be this:
![img](https://i.imgur.com/k0tSGW3.png)
